### PR TITLE
Un-doc functions taking debug_synchronous

### LINKED ISSUE #2199: [DOC]: Don't show `debug_synchronous` CUB overloads in documentation pages
### Is this a duplicate?

- [X] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Is this for new documentation, or an update to existing docs?

Update

### Describe the incorrect/future/missing documentation

The documentation for CUB device-scope algorithms is potentially confusing because it shows the signature of the overload that takes a `debug_synchronous` argument, but without any of the accompanying documentation. 

For example, the normal `DeviceReduce` overload looks like:
image
https://nvidia.github.io/cccl/cub/api/structcub_1_1DeviceReduce.html#_CPPv4I000EN3cub12DeviceReduce3SumE11cudaError_tPvR6size_t14InputIteratorT15OutputIteratorT9NumItemsT12cudaStream_t

But the one that takes `debug_synchronous` looks like:
image
https://nvidia.github.io/cccl/cub/api/structcub_1_1DeviceReduce.html#_CPPv4I00EN3cub12DeviceReduce3SumE11cudaError_tPvR6size_t14InputIteratorT15OutputIteratorTi12cudaStream_tb

It would be better to just exclude the overloads that take `debug_synchronous` from the documentation page all together. 

### If this is a correction, please provide a link to the incorrect documentation. If this is a new documentation request, please link to where you have looked.

_No response_

### PR DESCRIPTION
**Pull request by bryevdv**
fixes #2199 

This PR wraps all functions that accept `debug_synchronous` with 
```c
#ifndef DOXYGEN_SHOULD_SKIP_THIS // Do not document

#endif
```
in order to suppress them in the generated documentation. Search results in the local docs now only report the `.cuh` source files themselves as results, and not any API docs. 

## Before (currently deployed)

image

## After (local docs)

image

### CODE REVIEW

File: cub/cub/device/dispatch/dispatch_histogram.cuh [hunk 1]
```diff
@@ -1586,6 +1592,7 @@ public:
     return error;
   }
 
+#ifndef DOXYGEN_SHOULD_SKIP_THIS // Do not document
   CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t DispatchEven(
```
Comment (bernhardmgruber) [inline cub/cub/device/dispatch/dispatch_histogram.cuh:1596]: Unrelated: this overload is missing a `CUB_DETAIL_RUNTIME_DEBUG_SYNC_IS_NOT_SUPPORTED` annotation.

File: cub/cub/device/dispatch/dispatch_rle.cuh [hunk 2]
```diff
@@ -543,6 +543,7 @@ struct DeviceRleDispatch
     return error;
   }
 
+#ifndef DOXYGEN_SHOULD_SKIP_THIS // Do not document
   CUB_RUNTIME_FUNCTION _CCCL_FORCEINLINE static cudaError_t Dispatch(
```
Comment (bernhardmgruber) [inline cub/cub/device/dispatch/dispatch_rle.cuh:547]: Unrelated: missing `CUB_DETAIL_RUNTIME_DEBUG_SYNC_IS_NOT_SUPPORTED`.

**Review by bernhardmgruber — APPROVED** [review]
Thanks for the nice work! I will send a separate PR to add the two missing `CUB_DETAIL_RUNTIME_DEBUG_SYNC_IS_NOT_SUPPORTED` annotations that you made me discover!

File: cub/cub/device/device_adjacent_difference.cuh [hunk 3]
```diff
@@ -282,6 +283,7 @@ public:
 
     return SubtractLeftCopy(d_temp_storage, temp_storage_bytes, d_input, d_output, num_items, difference_op, stream);
   }
+#endif
```
Comment (miscco) [inline cub/cub/device/device_adjacent_difference.cuh:286]: We strife to not leave `#endif` without a matching comment, as it often is hard to see what branch this ends. I am pushing a comment

**Review by miscco — APPROVED** [review]
Thanks a lot for cleaning up the docs.

One minor nitpick that I am addressing myself

**Pull request merged**